### ISSUE TITLE
How translate error messages?

### ISSUE THREAD

**Issue by AxelBriche** [SymfonyCasts/reset-password-bundle#125]
I have this message:
> "You have already requested a reset password email. Please check your email or try again soon"

How can I translate that please?
This error message is not inside a translation file.

Thanks.

**Comment by kbond** [comment]
You would have to translate that message yourself and add to your translation file(s). Then call `$translator->trans($e->getReason());` before showing the message (as a flash or in a template).

**Comment by weaverryan** [comment]
@kbond is exactly correct! In your controller (I'm assuming you used MakerBundle to generate your setup), in a try/catch you are passed a `ResetPasswordExceptionInterface` exception. Each of these has a `->getReason()` method. Translate this in your controller before displaying it. This text will not change in future versions of this bundle (if it did, it would be a backwards compatibility break).

This is probably something we should add to the docs.

**Comment by bastien70** [comment]
Hello,

I would also like to do translations. I read your messages correctly, but I'm not sure I understood everything. I have never done a translation before. By helping me with the doc ', I therefore did this:

**config/packages/translation.yaml:**

```yaml
framework:
  default_locale: fr
  translator:
    default_path: "%kernel.project_dir%/translations"
    fallbacks:
      - fr
```

**translations/ResetPasswordBundle.fr.yaml:**

```yaml
You have already requested a reset password email. Please check your email or try again soon.: Vous avez déjà demandé un e-mail de réinitialisation du mot de passe. Veuillez vérifier votre e-mail ou réessayer bientôt.
The link in your email is expired. Please try to reset your password again.: Le lien dans votre e-mail a expiré. Veuillez réessayer de réinitialiser votre mot de passe.
Please update the request_password_repository configuration in config/packages/reset_password.yaml to point to your "request password repository` service.: Veuillez mettre à jour la configuration de request_password_repository dans config/packages/reset_password.yaml pour pointer vers votre service "request password repository".
The reset password link is invalid. Please try to reset your password again.: Le lien de réinitialisation du mot de passe n'est pas valide. Veuillez réessayer de réinitialiser votre mot de passe.
```

**src/Controller/ResetPasswordController:**

```php

//...
        try {
            $resetToken = $this->resetPasswordHelper->generateResetToken($user);
        } catch (ResetPasswordExceptionInterface $e) {
            $this->addFlash('reset_password_error', sprintf(
                'Un problème est survenu lors du traitement de votre demande de réinitialisation de mot de passe - %s',
                $translator->trans($e->getReason())
            ));

            return $this->redirectToRoute('app_forgot_password_request');
        }
```

I did a cache:clear to be sure. But the message is still in english.
I tried to add in my translations file this : 

```yaml
hello world!: coucou le monde !
```

and in my controller : 

```php
dd($translator->trans("hello world!"));
```

But same thing, it is still in english

**Comment by jrushlow** [comment]
There are PRs open in the Symfony Demo repository that demonstrate how this is done. I'm hoping to get some time to head over there and put the finishing touches on those PR's soon.

**Comment by codedmonkey** [comment]
Hi there, would it be useful to add these strings to the translation files of the bundle?

**Comment by kbond** [comment]
@codedmonkey, is [this](https://github.com/SymfonyCasts/reset-password-bundle/tree/main/src/Resources/translations) what you mean?

**Comment by jrushlow** [comment]
https://github.com/SymfonyCasts/reset-password-bundle/blob/main/src/Resources/translations/ResetPasswordBundle.en.xlf

**Comment by bocharsky-bw** [comment]
Hey guys!

I suggested translating error messages in #206 that was already merged 🎉  - but we still need to merge related PR in Maker, see https://github.com/symfony/maker-bundle/pull/1059 . Please, give it a try and review if you want! Also, feel free to help translating error messages to other languages, currently it's translated into en, ru and uk: https://github.com/SymfonyCasts/reset-password-bundle/blob/d7f9fdacb1a6c192f9cf2295754523e059a8cda3/src/Resources/translations/ResetPasswordBundle.en.xlf#L25-L48

Btw, for now you can easily translate those messages yourself if needed, find a workaround here: https://github.com/SymfonyCasts/reset-password-bundle/issues/202#issuecomment-1024019601

**Comment by bocharsky-bw** [comment]
I'm closing this - https://github.com/symfony/maker-bundle/pull/1059 is merged and Maker v1.37.0 is released. Now this bundle only needs missing translations. If you already generated reset password code with Maker before - please, update it according to the new changes in Maker, see https://github.com/symfony/maker-bundle/pull/1059 for the help
